flaskr/templates/blog/index.html

```diff
@@ -18,11 +18,24 @@ <h1>{{ post['title'] }}</h1>
         {% if g.user['id'] == post['author_id'] %}
           <a class="action" href="{{ url_for('blog.update', id=post['id']) }}">Edit</a>
         {% endif %}
+        <button class="toggle-content" onclick="toggleContent(this)">Expand</button>
       </header>
-      <p class="body">{{ post['body'] }}</p>
+      <p class="body" style="display: none;">{{ post['body'] }}</p>
     </article>
     {% if not loop.last %}
       <hr>
     {% endif %}
   {% endfor %}
+  <script>
```

Comment:
**Issue**: The `toggleContent` function directly manipulates the `style` attribute to show/hide content. Consider using CSS classes to toggle visibility instead, which is more maintainable and allows for easier styling changes.

**Fix**: function toggleContent(button) {
  const body = button.parentElement.nextElementSibling;
  body.classList.toggle('hidden');
  button.textContent = body.classList.contains('hidden') ? 'Expand' : 'Collapse';
}

// Add this CSS class
.hidden {
  display: none;
}